### PR TITLE
Add planned enhancement: Find criteria

### LINKED ISSUE #225: [PE-D][Tester D] Cannot find events with space in name specifically
If I know the event I want is two words I cannot fundEvent for a space character only.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: ji-just-ji/ped#4

### LINKED ISSUE #235: [PE-D][Tester B] Suggestions to feature find
**Current implementation of find**

If we have two members: Wen Xin and Hui Xing.

`findMember Xin` will only return member Wen Xin

**I suggests that:** the rule of finding can be less strict; 

for example, the command `findMember Xin` can return both Wen Xin and Hui Xing.

It may benefits users as users may remember members name wrongly. 

It will be useful if the find can return members whose name only contains part of the keyword.

**After adding command below**, there is `0 members listed!`:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: KumChaiYin/ped#8

### PR DESCRIPTION
**Pull request by dylkaw**
Closes #235 and closes #225.

What have you done in this PR?
* Added planned enhancements to `findMember` and `findEvent` commands to find all matches of given keywords.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -450,6 +450,21 @@ We propose to make the index error messages more specific and highlight to the u
 that index is wrong. For example, in the `enrol m/1 e/-1` input, we will show an error message to the user along 
 the lines of "The provided Event Index is not a non-zero unsigned integer."
 
+### Improve find criteria
```
Comment (SelwynAng) [inline docs/DeveloperGuide.md]: Suggestion to change this to Improve `findMember` and `findEvent` criteria

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -450,6 +450,21 @@ We propose to make the index error messages more specific and highlight to the u
...
+**Current Implementation:**
+* **Current Issue:** As of now, `findMember` and `findEvent` commands will only match the respective member and event if
+the name of the member or event contains a word that matches exactly the entire given find keyword. This means that the user will
+have to remember the entire name of the `Member` or `Event` to find it, reducing the effectiveness of `findMember` and `findEvent`.
+* **Example:** `findMember alex` will list a member with name "Alex Yeoh", but `findMember ale` will result in no member being
+listed.
+
+**Proposed solution:**
```
Comment (SelwynAng) [inline docs/DeveloperGuide.md]: Suggestion to captialise the 's' in solution

**Review by SelwynAng — APPROVED** [review]
LGTM aside from some formatting nitpicks

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dylkaw) [reply docs/DeveloperGuide.md]: Done!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (dylkaw) [reply docs/DeveloperGuide.md]: Done!

**Review by wr1159 — APPROVED** [review]
LGTM now, okay to merge once merge conflict is fixed!

**Comment by dylkaw** [comment]
> LGTM now, okay to merge once merge conflict is fixed!

Merge conflict has been fixed!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -478,6 +478,22 @@ We propose to make the index error messages more specific and highlight to the u
...
+have to remember the entire name of the `Member` or `Event` to find it, reducing the effectiveness of `findMember` and `findEvent`.
+* **Example:** `findMember alex` will list a member with name "Alex Yeoh", but `findMember ale` will result in no member being
+listed.
+
+**Proposed Solution:**
+
+We propose that the find criteria will be improved such that it will find any event/member name which contains the given keyword string,
+i.e. member/event name will not have to contain a word that matches exactly with the given find keyword. For example,
+the command `findMember a` will now match members with name "Alex", "Alice", etc. In addition, if the user remembers that
+an event name consists of 2 words, they can do `findEvent ‎`, which would list all events which have a blank space in
```
Comment (andrechuakj) [inline docs/DeveloperGuide.md]: Correct me if I'm wrong, but there should be 2 spaces after the `findEvent` command right?
```suggestion
an event name consists of 2 words, they can do `findEvent  ‎`, which would list all events which have a blank space in
```

**Review by ph-nathan — APPROVED** [review]
LGTM, same comment as Andre, and while you're here can you capitalize the s in line 475 "Proposed solution" as well? :) tho its not part of ur PR

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -472,12 +472,28 @@ user cannot tell immediately which provided index is wrong.
...
 There is a lack of information shown to the user which specific index is wrong.
 
-**Proposed solution:**
+**Proposed Solution:**
 
 We propose to make the index error messages more specific and highlight to the user which index is wrong and why
 that index is wrong. For example, in the `enrol m/1 e/-1` input, we will show an error message to the user along 
 the lines of "The provided Event Index is not a non-zero unsigned integer."
 
+### Improve findMember/findEvent criteria
```
Comment (andrechuakj) [inline docs/DeveloperGuide.md]: I think we should put these in code block:
```suggestion
### Improve `findMember`/`findEvent` criteria
```

**Pull request merged**